src/modules/flow/iio/nodes.c

```diff
@@ -645,22 +653,22 @@ color_reader_cb(void *data, struct sol_iio_device *device)
     struct sol_flow_node *node = data;
     struct color_data *mdata = sol_flow_node_get_private_data(node);
     struct sol_rgb out = {
-        .red_max = mdata->out_range.max,
-        .green_max = mdata->out_range.max,
-        .blue_max = mdata->out_range.max
+        .red_max = mdata->iio_config.out_range.max,
+        .green_max = mdata->iio_config.out_range.max,
+        .blue_max = mdata->iio_config.out_range.max
     };
```

Comment:
Thanks @edersondisouza ! I just changed the color data realted, and did not finish the modifications on other sensors. I will apply your suggestions: Using double for the RGB scale/offset